src/TestLogger/Core/TestCaseNameParser.cs

```diff
@@ -54,175 +54,41 @@ public static ParsedName Parse(string fullyQualifiedName)
-                        {
-                            // If we found the beginning of the parenthesis block, we are back in
-                            // default state
-                            state = NameParseState.Default;
-                        }
-                        else if (thisChar == '"')
-                        {
-                            // This must come at the end of a string, when escape characters aren't
-                            // an issue, so we are 'entering' string state, because of the reverse parsing.
-                            state = NameParseState.String;
-                        }
-                        else if (thisChar == '\'')
-                        {
-                            state |= NameParseState.Char;
-                        }
-
-                        output.Insert(0, thisChar);
-                    }
-                    else if (state == NameParseState.String)
-                    {
-                        if (thisChar == '"' && fullyQualifiedName.ElementAtOrDefault(i - 1) != '\\')
-                        {
-                            // If this is a quote that has not been escaped, switch the state. If it
-                            // had been escaped, we would still be in a string.
-                            state = NameParseState.Parenthesis;
-                        }
-
-                        output.Insert(0, thisChar);
-                    }
-                    else if (state == NameParseState.Char)
-                    {
-                        if (thisChar == '\'' && fullyQualifiedName.ElementAtOrDefault(i - 1) != '\\')
-                        {
-                            // If this is a single quote that has not been escaped, switch the state. If it
-                            // had been escaped, we would still be in a char.
-                            state = NameParseState.Parenthesis;
-                        }
-
-                        output.Insert(0, thisChar);
-                    }
-                }
+                // Occassionally we get multi line results. Flatten those.
+                fullyQualifiedName = fullyQualifiedName.Replace("\r", string.Empty);
+                fullyQualifiedName = fullyQualifiedName.Replace("\n", string.Empty);
 
-                if (parenthesisCount != 0)
-                {
-                    throw new Exception($"Unbalanced count of parentheses found ({parenthesisCount})");
-                }
+                var regexOptions = RegexOptions.Compiled | RegexOptions.IgnoreCase;
 
-                // We are done. If we are finding type, set that variable. Otherwise, there was some
-                // issue, so leave the type blank.
-                if (step == NameParseStep.FindNamespace)
+                var method = new Regex(@"^([a-z1-9_.]{1,})\.([a-z1-9_.]{1,})\.(.{1,})$", regexOptions); // This one picks up anything with just method parameters or no method params.
```

Comment:
Creating `Regex` objects is expensive. Can we consider making this instance and reuse? Or, consider using the static `Regex.Match` methods?

Reply:
Made static versions.